server/application/application.go

```diff
@@ -289,13 +289,24 @@ func (s *Server) Update(ctx context.Context, q *ApplicationUpdateRequest) (*appv
 // drops old overrides that are invalid
 // throws an error is passed override is invalid
 // if passed override and old overrides are invalid, throws error, old overrides not dropped
-func (s *Server) removeInvalidOverrides(a *appv1.Application, q *ApplicationUpdateSpecRequest) (*ApplicationUpdateSpecRequest, error) {
+func (s *Server) removeInvalidOverrides(ctx context.Context, a *appv1.Application, q *ApplicationUpdateSpecRequest) (*ApplicationUpdateSpecRequest, error) {
```

Comment:
Honestly, I think we can probably remove removeInvalidOverrides function. I don't think it works for helm, and the logic confuses me. It's up to you.

Reply:
I agree. Removing it.